src/api-to-friendly.ts

```diff
@@ -24,10 +24,22 @@ const apiToFriendlyRelation = (
   relationDefinition: Userset,
   relations: string[],
   idx: number,
+  metadata: Userset | Metadata | undefined,
   newSyntax: string[],
 ) => {
   const relationKeys = Object.keys(relationDefinition);
-  const define = [`    ${Keywords.DEFINE} ${relation}${relationKeys?.length ? ` ${Keywords.AS} ` : ""}`];
+  const relationMetadata = (metadata as Metadata)?.relations?.[relation];
+  
+  let allowedTypesArray = relationMetadata?.directly_related_user_types?.map((relationReference) => {
```

Comment:
```suggestion
  const allowedTypesArray = relationMetadata?.directly_related_user_types?.map((relationReference) => {
```

Are we changing allowedTypesArray?  If not, use const